File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <iostream>
#include <algorithm>

using namespace std;

int binarySearch(const int a[], int n, int key);

int main(void)
{
    ios_base::sync_with_stdio(0); 
    cin.tie(0);
    
	int inputNum;
	cin >> inputNum;

	int* arr = new int[inputNum];
	for (int i = 0; i < inputNum; i++) {
		cin >> arr[i];
	}
	sort(arr, arr + inputNum);

	int compNum;
	cin >> compNum;

	for (int i = 0; i < compNum; i++) {
		int comp;
		cin >> comp;

		if (binarySearch(arr, inputNum, comp) >= 0) printf("1\n");
		else printf("0\n");
	}

	return 0;
}

int binarySearch(const int a[], int n, int key)
{
	int left = 0, right = n - 1;

	do {
		int center = (left + right) / 2;
		if (a[center] == key) return center;
		else if (a[center] < key) left = center + 1;
		else right = center - 1;
	} while (left <= right);

	return -1;
}

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.util.Scanner;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        int[] A = new int[n];
        for (int i = 0; i < n; i++) {
            A[i] = scanner.nextInt();
        }
        
        int m = scanner.nextInt();
        
        Arrays.sort(A);
        
        for (int i = 0; i < m; i++) {
            int target = scanner.nextInt();
            if (binarySearch(A, target) == 1) {
                System.out.println(1);
            } else {
                System.out.println(0);
            }
        }
    }
    
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;
        
        while (left <= right) {
            int mid = (left + right) / 2;
            if (target == arr[mid]) {
                return 1;
            } else if (target < arr[mid]) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return -1;
    }
}